File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/AbpUnitOfWorkOptions.java
package com.ueueo.uow;

import lombok.Getter;
import lombok.Setter;

/**
 * @author Lee
 * @date 2022-05-19 20:35
 */
@Getter
@Setter
public class AbpUnitOfWorkOptions implements IAbpUnitOfWorkOptions {
    private boolean isTransactional;
    private IsolationLevel isolationLevel;
    private Integer timeout;

    public AbpUnitOfWorkOptions() {}

    public AbpUnitOfWorkOptions(Boolean isTransactional, IsolationLevel isolationLevel, Integer timeout) {
        this.isTransactional = isTransactional != null ? isTransactional : false;
        this.isolationLevel = isolationLevel;
        this.timeout = timeout;
    }

    @Override
    public AbpUnitOfWorkOptions clone() {
        return new AbpUnitOfWorkOptions(isTransactional, isolationLevel, timeout);
    }

}

File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/AbpUnitOfWorkOptions.cs
using System;
using System.Data;

namespace Volo.Abp.Uow;

public class AbpUnitOfWorkOptions : IAbpUnitOfWorkOptions
{
    /**
     * Default: false.
    */
    public boolean IsTransactional;// { get; set; }

    public IsolationLevel? IsolationLevel;// { get; set; }

    /**
     * Milliseconds
    */
    public int? Timeout;// { get; set; }

    public AbpUnitOfWorkOptions()
    {

    }

    public AbpUnitOfWorkOptions(boolean isTransactional = false, IsolationLevel? isolationLevel = null, int? timeout = null)
    {
        IsTransactional = isTransactional;
        IsolationLevel = isolationLevel;
        Timeout = timeout;
    }

    public AbpUnitOfWorkOptions Clone()
    {
        return new AbpUnitOfWorkOptions
        {
            IsTransactional = IsTransactional,
            IsolationLevel = IsolationLevel,
            Timeout = Timeout
        };
    }
}
